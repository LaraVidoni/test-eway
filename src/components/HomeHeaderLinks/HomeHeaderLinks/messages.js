export default {
    en: {
      home: "Home",
      logout: "Logout",
      aboutUs: "About Us",
      language: "Français",
      contact: "For Support Contact ",
      defaultContactAddress: "easyservice@staples.com"
    },
    fr: {
      home: "Accueil",
      logout: "Déconnexion",
      aboutUs: "À propos",
      language: "English",
      contact: "Pour assistance : ",
      defaultContactAddress: "servicefacile@staples.com"
    }
  };
  