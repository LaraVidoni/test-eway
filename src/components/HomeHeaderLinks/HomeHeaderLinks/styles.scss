.App {
    text-align: Left;
  }
  
  .App a {
    text-decoration: Underline;
    color: black;
    margin: 0 5px 0 5px;
  }
  