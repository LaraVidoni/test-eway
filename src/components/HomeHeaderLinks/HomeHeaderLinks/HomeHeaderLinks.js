import React from "react";
import messages from "./messages";
import "./styles.scss";

function HomeHeaderLinks(props) {
  const {
    showLogoff = true,
    showAboutUs = true,
    showLanguage = true,
    contactAddress = "easyservice@staples.com",
    language = "en"
  } = props;
  let resources = messages[language];

  return (
    <div className="HeaderLinkComponent">
      {
        <HeaderLinkItem url={`${window.location.href}`}>
          {resources.home}
        </HeaderLinkItem>
      }
      {showLogoff && (
        <HeaderLinkItem url={`logoff.aspx`}>{resources.logout}</HeaderLinkItem>
      )}
      {showAboutUs && (
        <HeaderLinkItem url={`AboutUs.aspx`}>
          {resources.aboutUs}
        </HeaderLinkItem>
      )}
      {showLanguage && (
        <HeaderLinkItem
          url={`${language}/Core/Home/SetCulture?previousSearchParameters=${language}`}
        >
          {resources.language}
        </HeaderLinkItem>
      )}
    </div>
  );
}

function HeaderLinkItem(props) {
  return (
    <span className="HeaderLinkItem">
      <a href={props.url}>{props.children}</a>
    </span>
  );
}

export default HomeHeaderLinks;
