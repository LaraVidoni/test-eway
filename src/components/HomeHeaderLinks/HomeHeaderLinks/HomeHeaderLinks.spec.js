/* eslint-disable filenames/match-regex */
import React from "react";
import HomeHeaderLinks from "./HomeHeaderLinks";
import renderer from "react-test-renderer";

it(`hides the "about us"`, () => {
  const state = {
    showAboutUs: false,
  }
  const tree = renderer.create(<HomeHeaderLinks {...state} />).toJSON();
  expect(tree).toMatchSnapshot();
});
it(`hides the "logoff"`, () => {
  const state = {
    showLogoff: false
  }
  const tree = renderer.create(<HomeHeaderLinks {...state} />).toJSON();
  expect(tree).toMatchSnapshot();

});
it(`hides the "language switcher"`, () => {
  const state = {
    showLanguage: false,
  }
  const tree = renderer.create(<HomeHeaderLinks {...state} />).toJSON();
  expect(tree).toMatchSnapshot();
});
it(`shows the "English" word when in french`, () => {
  const state = {
    language: 'fr',
  }
  const tree = renderer.create(<HomeHeaderLinks {...state} />).toJSON();
  expect(tree).toMatchSnapshot();
});