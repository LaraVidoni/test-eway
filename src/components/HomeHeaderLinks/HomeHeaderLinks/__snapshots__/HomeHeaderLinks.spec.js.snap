// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides the "about us" 1`] = `
<div
  className="HeaderLinkComponent"
>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="http://localhost/"
    >
      Home
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="logoff.aspx"
    >
      Logout
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="en/Core/Home/SetCulture?previousSearchParameters=en"
    >
      Français
    </a>
  </span>
</div>
`;

exports[`hides the "language switcher" 1`] = `
<div
  className="HeaderLinkComponent"
>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="http://localhost/"
    >
      Home
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="logoff.aspx"
    >
      Logout
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="AboutUs.aspx"
    >
      About Us
    </a>
  </span>
</div>
`;

exports[`hides the "logoff" 1`] = `
<div
  className="HeaderLinkComponent"
>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="http://localhost/"
    >
      Home
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="AboutUs.aspx"
    >
      About Us
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="en/Core/Home/SetCulture?previousSearchParameters=en"
    >
      Français
    </a>
  </span>
</div>
`;

exports[`shows the "English" word when in french 1`] = `
<div
  className="HeaderLinkComponent"
>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="http://localhost/"
    >
      Accueil
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="logoff.aspx"
    >
      Déconnexion
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="AboutUs.aspx"
    >
      À propos
    </a>
  </span>
  <span
    className="HeaderLinkItem"
  >
    <a
      href="fr/Core/Home/SetCulture?previousSearchParameters=fr"
    >
      English
    </a>
  </span>
</div>
`;
